From f2edc4f9262e9130d020ef6caef2443e7ae31371 Mon Sep 17 00:00:00 2001
From: Eduardo Habkost <ehabkost@redhat.com>
Date: Wed, 12 Aug 2020 10:58:04 +0200
Subject: q35: Set max_cpus to 512

RH-Author: Eduardo Habkost <ehabkost@redhat.com>
Message-id: <20200729180236.627559-5-ehabkost@redhat.com>
Patchwork-id: 98076
O-Subject: [RHEL-AV-8.3.0 qemu-kvm PATCH v2 4/4] q35: Set max_cpus to 512
Bugzilla: 1819292
RH-Acked-by: Andrew Jones <drjones@redhat.com>
RH-Acked-by: Thomas Huth <thuth@redhat.com>
RH-Acked-by: David Gibson <dgibson@redhat.com>

Increase supported VCPU count for the Q35 machine type.

The VCPU count that partners confirmed to work depended on other
parameters (especially RAM size), but fluctuated between 640 and
710 VCPUs.  I chose to increase the limit to 512 to be
conservative, until we find out what exactly prevents larger VMs
from booting.

Signed-off-by: Eduardo Habkost <ehabkost@redhat.com>
Signed-off-by: Miroslav Rezanina <mrezanin@redhat.com>
Signed-off-by: Danilo C. L. de Paula <ddepaula@redhat.com>
---
 hw/i386/pc_q35.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/hw/i386/pc_q35.c b/hw/i386/pc_q35.c
index d75d6d8805..c709460ab7 100644
--- a/hw/i386/pc_q35.c
+++ b/hw/i386/pc_q35.c
@@ -585,7 +585,7 @@ static void pc_q35_machine_rhel_options(MachineClass *m)
     machine_class_allow_dynamic_sysbus_dev(m, TYPE_INTEL_IOMMU_DEVICE);
     machine_class_allow_dynamic_sysbus_dev(m, TYPE_RAMFB_DEVICE);
     m->alias = "q35";
-    m->max_cpus = 384;
+    m->max_cpus = 512;
     compat_props_add(m->compat_props, pc_rhel_compat, pc_rhel_compat_len);
 }
 
-- 
2.27.0

